flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -2624,7 +2624,7 @@ private SqlNode registerFrom(
                                 usingScope,
                                 register,
                                 operand,
-                                enclosingNode,
+                                call,
```

Comment:
@luoyuxia  I modify this as  time travel with as expression not work well .
SELECT * FROM test_tb FOR SYSTEM_TIME AF OF  TIMESTAMP '2023-01-01 01:00' as t2  

If I don't modity this,  in FlinkCalciteSqlValidator:registerNamespace method, ns.getEnclosingNode will not be SqlSnapshot.

Reply:
Make sense to me, I have rolled back this logic.